File: security-svc/src/main/java/com/chilly/security_svc/repository/UserRepository.java
package com.chilly.security_svc.repository;

import com.chilly.security_svc.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findByEmail(String email);

    Optional<User> findByPhoneNumber(String phoneNumber);
}

File: business-users-svc/src/main/kotlin/org/chilly/business/users/repository/UserRepository.kt
package org.chilly.business.users.repository

import org.chilly.business.users.model.BusinessUser
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<BusinessUser, Long> {

    fun findByCompanyName(companyName: String): BusinessUser?

    fun findByInn(inn: String): BusinessUser?

}